includes/utils.php

```diff
@@ -144,7 +144,9 @@ public static function check_cache( $args ) {
 			}
 		}
 
-		self::set_cache( $args['cache_key'], $results, $args['group'], $args['time'] );
+		if ( ! empty( $results ) ) {
```

Comment:
Hi @sorinmarta,

Thanks for your work on this.

We should move the validation inside the `self::set_cache` method as it's more future-proof and will offer safer usage elsewhere in the future.

Reply:
Thanks for the review @shervElmi! I didn't expect you to be so fast. 😄

While you were reviewing I was thinking about it and realized that checking for empty for every kind of usage may cause a performance impact for things that should be stored as empty.

Like an empty category for example. It will always run the query and could definitely be optimized with cache.

So I decided to switch to a parameter that can then be used only where empty values should not be stored.

Good point with moving that to `set_cache` as that's also a public method. I'll change it to that.

Thanks!